lib/chef/guard_interpreter/resource_guard_interpreter.rb

```diff
@@ -68,7 +68,8 @@ def evaluate_action(action=nil, &block)
         run_action = action || @resource.action
 
         begin
-          @resource.run_action(run_action)
+          # Coerce to an array to be safe.
```

Comment:
Resources that override `action` or `default_action`, or that set `@action` manually in an old-school `initialize`.